pkg/doc/verifiable/credential.go

```diff
@@ -445,15 +446,63 @@ func (i *Issuer) UnmarshalJSON(bytes []byte) error {
 }
 
 // Subject of the Verifiable Credential
-type Subject interface{}
+type Subject struct {
+	ID string `json:"id,omitempty"`
+
+	CustomFields CustomFields `json:"-"`
+}
+
+// MarshalJSON marshals Subject to JSON.
+func (s Subject) MarshalJSON() ([]byte, error) {
+	if len(s.CustomFields) == 0 {
+		// Subject ID as string
+		return json.Marshal(s.ID)
+	}
+
+	type Alias Subject
+
+	alias := Alias(s)
+
+	data, err := marshalWithCustomFields(alias, s.CustomFields)
+	if err != nil {
+		return nil, fmt.Errorf("marshal Subject: %w", err)
+	}
+
+	return data, nil
+}
+
+// UnmarshalJSON unmarshals Subject from JSON.
+func (s *Subject) UnmarshalJSON(bytes []byte) error {
+	var subjectID string
+
+	if err := json.Unmarshal(bytes, &subjectID); err == nil {
+		// as string
+		s.ID = subjectID
+		return nil
+	}
+
+	type Alias Subject
+
+	alias := (*Alias)(s)
+
+	s.CustomFields = make(CustomFields)
+
+	err := unmarshalWithCustomFields(bytes, alias, s.CustomFields)
+	if err != nil {
+		return fmt.Errorf("unmarshal Subject: %w", err)
+	}
+
+	return nil
+}
 
 // Credential Verifiable Credential definition
 type Credential struct {
-	Context        []string
-	CustomContext  []interface{}
-	ID             string
-	Types          []string
-	Subject        Subject
+	Context       []string
+	CustomContext []interface{}
+	ID            string
+	Types         []string
+	// Subject can be a string, map, slice of maps, struct (Subject or any custom), slice of structs.
```

Comment:
can you show an example of where it's just a string?

Reply:
According to [VC data model documentation](https://www.w3.org/TR/vc-data-model/#credential-subject)
> Each object MAY contain an id
so according to JSON-LD spec, if the JSON object contains just `id` field, it can be simplified to just a string containing that id.